compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -489,8 +489,30 @@ class OMR_EXTENSIBLE CodeGenerator
    bool isRegisterClobberable(TR::Register *reg, uint16_t count);
 
    // ilgen
-   bool ilOpCodeIsSupported(TR::ILOpCodes); // no virt, default, cast
-
+   /**
+    * @brief Returns if an IL OpCode is supported by current CodeGen
+    *
+    * @param op The IL OpCode being checked.
+    *
+    * @return True if the IL OpCode is supported otherwise false.
+    */
+   static bool ilOpCodeIsSupported(TR::ILOpCodes op); // no virt, default, cast
+   /**
+    * @brief Returns the corresponding IL OpCode for an intrinsic function
+    *
+    * @param op The intrinsic function being checked.
+    *
+    * @return The corresponding IL OpCode for the intrinsic function.
+    */
+   static TR::ILOpCodes ilOpCodeForIntrinsicFunction(TR::RecognizedMethod m) { return TR::BadILOp; }// no virt, default, cast
```

Comment:
This doesn't sound like an OMR type query since its very downstream dependent. What is the envisioned use for this API?

Reply:
The concept of intrinsic function exists in most if not all programming languages, to expose features that is not trivial in the language itself, for example, parallelization, complex mathematics, and synchronization. OMR providing this query can help in two aspects: 1. providing a unified framework for downstream projects to better organize their intrinsic functions; 2. giving Optimizer a better idea about an intrinsic function.

Reply:
`suppressInliningOfRecognizedMethod()` is a perfect example for using this query so that I've updated the change to include `suppressInliningOfRecognizedMethod()`.

Reply:
Ok, sounds reasonable. One comment though in that to my knowledge for symmetry we seem to prefer to use "method" rather than "function" in OMR, i.e. `RecognizedMethod`, `OMRMethod`, `TrustedMethod`, `MethodInliner`, etc. To stay consistent what do you think about renaming this API to `ilOpCodeForIntrinsicMethod` to match the type name of it's only argument?